src/wagtail_ai/views.py

```diff
@@ -98,14 +106,70 @@ def process(request) -> JsonResponse:
     try:
         response = handler(prompt=prompt, text=prompt_form.cleaned_data["text"])
     except AIHandlerException as e:
-        return JsonResponse({"error": str(e)}, status=400)
+        return ErrorJsonResponse(str(e), status=400)
     except Exception:
         logger.exception("An unexpected error occurred.")
-        return JsonResponse({"error": _("An unexpected error occurred.")}, status=500)
+        return ErrorJsonResponse(_("An unexpected error occurred."))
 
     return JsonResponse({"message": response})
 
 
+def user_has_permission_for_image(user, image):
+    from wagtail.images.permissions import permission_policy
+
+    return permission_policy.user_has_permission_for_instance(user, "choose", image)
+
+
+def describe_image(request) -> JsonResponse:
```

Comment:
**Question:** Would there ever be a use case of limited access to the AI functionality for this? Should there be a custom permission?

Reply:
The current code assumes that if you can choose an image to include on a page, you can also describe it. You're rather asking about restricting access _to the AI functionality_, no? That goes hand in hand with audit logging (as you also mention). My gut feeling is that it's a separate task that merits a new ticket, because it should also address the text completion functionality.

Reply:
Raised #85.